### DISCUSSION TITLE
Tailwind classes/style overwritten

### DISCUSSION THREAD

**Discussion by fbobbio** [ng-icons/ng-icons#159] [Q&A] [answered]
Hello all 👋 

I've started using the library, I must admit i suck at css/styling :)

But migrating from another angular icons lib to `ng-icons` I noticed some specific classes are overwritten/ignored by the component.

Example:

`<ng-icon class="h-4 w-4 text-slate-800 group-hover:text-red-500" name="trash"></ng-icon>`

on this ☝️ the `h-4`, `w-4` and `test-slate-800` will be overwritten by the default (even set or not set) size/color atts of ng-icons.

I might be missing something simple/fundamental.

But, any suggestion of how can I keep the classes priority without getting into `!important` use or similar approaches?
Ideally the element-defined classes are the ones rendered.

Thanks in advance

**Comment by ashley-hunter** [comment, marked as answer]
Hey! Apologies for the delay in getting back to you, took a few days away from the laptop over the holidays. Sure, so Tailwind should work absolutely fine with Ng Icons, in fact we use Tailwind in our docs site extensively!

From what I gather the main issue is the use of size classes for setting the dimensions, rather then w-* or h-* classes, icons use the font size to set their size, we do this as it allows icons to naturally fit into its context in most cases, for example if it is inline with some text it would appear the correct size, or perhaps used within a button etc... it also is a single value whereas width and height are 2 dimensions that would need to be defined rather than just one.

So the first thing to do would be to change the width classes to text size classes e.g.:

```html
<ng-icon class="text-base text-slate-800 group-hover:text-red-500" name="trash" />
```

The color classes should work as expected, particularly in more recent versions of NgIcons as we define the color in a separate CSS layer to allow user defined colors to always take precedence without needing to add `!important`.

Here is an example showing this:

https://stackblitz.com/edit/stackblitz-starters-tqw3jwx1?file=src%2Fmain.ts

If you find a case that isn't working, just let me know and I will happily look into it! Thank you!

**Reply by fbobbio** [reply]
Hey @ashley-hunter , thanks a LOT for your prompt reply and clear explanation.

It was lightning fast ⚡

Your answer helped me validate that the size approach you suggested is correct and then, looking a bit closer, i found we have a bit of a messy structure of CSS layers that's the reason why the tailwind text color classes were overwritten.

I still did not solve the issue, but now I have a few leads to follow.

I am on ng-icons `29.10.0` by the way :) 

Thanks and happy new year!

**Reply by ashley-hunter** [reply]
Great, glad that helped, if there's anything I can do to help just let me know! Thank you, and happy new year to you too!
